### PR TITLE
pytester: use temporary HOME with spawn

### PR DESCRIPTION
**Pull request by blueyed**
Followup to https://github.com/pytest-dev/pytest/issues/4956.

### CODE REVIEW

File: changelog/4956.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+pytester's ``testdir.spawn`` uses a temporary HOME directory.
```
Comment (nicoddemus) [inline changelog/4956.feature.rst]:
```suggestion
pytester's ``testdir.spawn`` uses ``tmpdir`` as HOME/USERPROFILE directory.
```

**Comment by blueyed** [comment]
Refactored it, which also allows for adjusting it via `testdir._env_run_update`.

**Comment by blueyed** [comment]
This would allow for the following now, which works around pdb++ hijacking pdb and adding source code highlighting, which fails some (3) tests currently.  Not sure if it should be added though, given that it is rather specific (pdbpp being installed), and tests could be adjusted to cope with it (i.e. loser/other assertions).

```diff
diff --git i/testing/test_pdb.py w/testing/test_pdb.py
index c0ba7159b..81e455b13 100644
--- i/testing/test_pdb.py
+++ w/testing/test_pdb.py
@@ -76,6 +76,13 @@ def set_trace(self, frame):
     del _pytest._CustomDebugger


+@pytest.fixture(autouse=True)
+def pdb_env(request):
+    if "testdir" in request.funcargnames:
+        testdir = request.getfixturevalue("testdir")
+        testdir._env_run_update["PDBPP_HIJACK_PDB"] = "0"
+
+
 class TestPDB(object):
     @pytest.fixture
     def pdblist(self, request):
```

**Comment by blueyed** [comment]
Note that the above fixture is different from setting `PDBPP_HIJACK_PDB=0` before running pytest: it would still use pdbpp then in the outer tests.

**Comment by nicoddemus** [comment]
> This would allow for the following now, which works around pdb++ hijacking pdb and adding source code highlighting, which fails some (3) tests currently. Not sure if it should be added though, given that it is rather specific (pdbpp being installed), and tests could be adjusted to cope with it (i.e. loser/other assertions).

I'm fine if this helps people with pdb++ installed to run the pytest suite (such as yourself), just please make sure to add an informative docstring to the fixture though.

Also feel free to merge this now if you prefer to introduce this fixture in a future PR instead.

**Comment by blueyed** [comment]
Test was skipped - needs https://github.com/pytest-dev/pytest/pull/5313.

File: tox.ini [hunk 2]
```diff
@@ -40,7 +40,7 @@ setenv =
     lsof: _PYTEST_TOX_POSARGS_LSOF=--lsof
 
     pexpect: _PYTEST_TOX_PLATFORM=linux|darwin
-    pexpect: _PYTEST_TOX_POSARGS_PEXPECT=testing/test_pdb.py testing/test_terminal.py testing/test_unittest.py
+    pexpect: _PYTEST_TOX_POSARGS_PEXPECT=-m uses_pexpect
```
Comment (blueyed) [inline tox.ini]: Switched to using the new marker here.

Comment (blueyed) [reply tox.ini]: Meh.. that will fail when combined with twisted etc I guess.
Also with using pexpect for macos in the other PR.
Should get back to https://github.com/pytest-dev/pytest/pull/5313 first...

Comment (blueyed) [reply tox.ini]: ...it is not being combined though, so that should be good.

Comment (blueyed) [reply tox.ini]: Pulled that into the fix-pexpect PR (6896dbc).

**Pull request merged**